Gillian-C/lib/gil_logic_gen.ml

```diff
@@ -155,7 +154,7 @@ let assert_of_member cenv members id typ =
     let n_e = Expr.int_z n in
     let chunk =
       match Ctypes.access_mode ty with
-      | By_value chunk -> chunk
+      | By_value chunk -> Chunk.of_compcert chunk
```

Comment:
You have the type here, if it's a pointer type you should be using the `Mptr` chunk